### PR TITLE
Support for attention request

### PR DESCRIPTION
**Pull request by felipefs**
This pull contains the implementation of attention request for X11.

The added function was ```glfwRequestWindowAttention```.

The example ```simple.c``` was updated to demonstrate the functionality.

This patch is related with the issue https://github.com/glfw/glfw/issues/732.

### CODE REVIEW

File: src/cocoa_window.m [hunk 1]
```diff
@@ -1272,6 +1272,11 @@ void _glfwPlatformShowWindow(_GLFWwindow* window)
     [window->ns.object orderFront:nil];
 }
 
+void _glfwPlatformRequestWindowAttention(_GLFWwindow* window)
+{
+    [window->ns.object requestUserAttention:NSInformationalRequest];
```
Comment (elmindreda) [inline src/cocoa_window.m]: `requestUserAttention:` is part of NSApplication, not NSWindow.

**Comment by elmindreda** [comment]
Because some platforms (read macOS) have application-level attention requests, perhaps the more neutral `glfwRequestAttention` would be a better name?

**Comment by felipefs** [comment]
I was wondering how we could approach that without breaking compatibility between platforms, since on Linux/Window, it will look like ```glfwRequestAttention(GLFWwindow *)``` and on macOS ```glfwRequestAttention(void)```.

**Comment by elmindreda** [comment]
Have it take a window and document that it's a hint and not guaranteed to point out the specific window.

**Comment by felipefs** [comment]
I've updated the function to do the following:
```objective-c
void _glfwPlatformRequestWindowAttention(_GLFWwindow* window)
{
    [NSApp requestUserAttention:NSInformationalRequest];
}
```
Also, I updated the documentation, adding a remark that on OSX the attention refers to the application, not the window.

Let me know if something is incorrect and I will fix 👍

**Comment by elmindreda** [comment]
This looks almost ready to merge. Please don't add any test code or additional features to the `simple` example, though. It's mean to be just that and is [used in the documentation](http://www.glfw.org/docs/latest/quick_guide.html#quick_example).

**Comment by felipefs** [comment]
Do you want me to remove the added code from ```simple.c```?

**Comment by elmindreda** [comment]
Yes, please do. Instead you could for example request attention whenever the `joysticks` test detects a connection or disconnection.

File: src/x11_window.c [hunk 2]
```diff
@@ -2078,6 +2078,21 @@ void _glfwPlatformHideWindow(_GLFWwindow* window)
...
+
+    memset(&xev, 0, sizeof(xev));
+    xev.type = ClientMessage;
+    xev.xclient.window = window->x11.handle;
+    xev.xclient.message_type = _glfw.x11.NET_WM_STATE;
+    xev.xclient.format = 32;
+    xev.xclient.data.l[0] = _NET_WM_STATE_ADD;
+    xev.xclient.data.l[1] = _glfw.x11.NET_WM_STATE_DEMANDS_ATTENTION;
+
+    XSendEvent(_glfw.x11.display, DefaultRootWindow(_glfw.x11.display), False, SubstructureRedirectMask | SubstructureNotifyMask, &xev);
```
Comment (elmindreda) [inline src/x11_window.c]: This function could be made significantly shorter by calling the helper `sendEventToWM`.

**Comment by elmindreda** [comment]
This has been merged as 412eb6a6113f71f42ad56d8dd982003ba7094870, with additional cleanup in baa9cd8968e1571cd10a7265b589d997febf611e and 2f5e2303380d084ab9b097765d34f98504324d65.

**Pull request closed without merging**